aesara/tensor/blockwise.py

```diff
@@ -0,0 +1,326 @@
+                return var
+
+            if var in core_inputs:
+                return inputs[core_inputs.index(var)]
+            if var in core_outputs:
+                return outputs[core_outputs.index(var)]
+            if var in core_out_grads:
+                return ograds[core_out_grads.index(var)]
+
+            node = var.owner
+            if node is None:
+                # The gradient contains a constant
+                # res = aesara.tensor.basic.constant(
+                #     np.asarray(var.data), dtype=var.type.dtype
+                # )
+                res = var
+
+                # TODO FIXME: Use dimensions of relevant/appropriate inputs.
+                # What exactly are those in this case?
+                nd = inputs[0].type.ndim
+
+                return atleast_Nd(res, nd)
+
+            blocked_inputs = [transform(ipt, node) for ipt in node.inputs]
+
+            grad_signature = getattr(node.op, "gufunc_sig", None)
+
+            if grad_signature is None:
+                if isinstance(node.op, DimShuffle):
+                    # remove the extra dimensions that
+                    # we have added during op creation
+                    new_order = [i for i in node.op.new_order if i != "x"]
+
+                    # derive gufunc signature for DimShuffle
+                    input_signature = tuple([f"a{i}" for i in range(len(new_order))])
+                    output_signature = tuple([f"a{i}" for i in new_order])
+                    grad_signature = ((input_signature,), (output_signature,))
+                else:
+                    raise ValueError(
+                        f"'{node.op}' object has no attribute 'gufunc_sig'"
+                    )
+
+            new_r = Blockwise(node.op, signature=grad_signature)(*blocked_inputs)
+            assert isinstance(new_r, Variable)
+            return new_r
+
+        ret = []
+        for core_inp_grad, ipt in zip(core_inp_grads, inputs):
+            ret.append(transform(core_inp_grad, None))
+
+        return ret
+
+    def perform(self, node, inputs, outputs):
+        def py_func(*inner_inputs):
+            res = [[None]] * len(outputs)
+            # TODO:This can be avoided by making a single dummy node
+            # But will that cover all cases?
+            inner_node = self.op.make_node(*inner_inputs)
+            if isinstance(self.op, DimShuffle):
+                self.op.perform(inner_node, inner_inputs, res, params=None)
+            else:
+                self.op.perform(inner_node, inner_inputs, res)
+
+            # Numpy always expects outputs to be Numpy arrays
+            # And since we have a variable number of outputs
+            if len(res) == 1:
+                return res[0][0]
+            else:
+                return tuple(_res[0] for _res in res)
+
+        numpy_vec_func = np.vectorize(
+            py_func, signature=gufunc_sign_to_str(self.signature)
+        )
+        res_variables = numpy_vec_func(*inputs)
```

Comment:
I am getting "ValueError: inconsistent size for core dimension" here but unable to figure out the solution

Reply:
Which test?

Reply:
for test_blockwise_solve_grad(), when a_shape is (2, 2) and b_sbape is (2, )